src/bramble/client/ProjectStats.js

```diff
@@ -0,0 +1,132 @@
+define([
+    // Change this to filer vs. filer.min if you need to debug Filer
+    "thirdparty/filer/dist/filer.min",
+], function(Filer){
+    "use strict";
+
+    var ProjectStats = {};
+    var _Path = Filer.Path;
+    var _root;
+
+    // keep track of project state
+    // host cache in the object should any other project states have to be added
+    var _cache = {};
+
+    var _fs = new Filer.FileSystem();
+    var _shell = new _fs.Shell();
+    ProjectStats.getFileSystem = function(){
+        return _fs;
+    };
+    // NOTE: THIS WILL DESTROY DATA! For error cases only, or to wipe the disk.
+    ProjectStats.formatFileSystem = function(flags, callback){
+        _fs = new Filer.FileSystem(flags, callback);
+        _shell = new _fs.Shell();
+        return _fs;
+    };
+
+    // walk the whole file tree and record each path and file size
+    ProjectStats.init = function(root, callback){
+        function addSize(path, next){
+            // ignore directories and do nothing
+            if (path.endsWith("/")){
+                return next();
+            }
+
+            _fs.stat(path, function(err, stats){
+                if (err){
+                    return next(err);
+                }
+                _cache[path] = stats.size;
+                next();
+            });
+        }
+
+        // original unlink
+        var _innerUnlink = _fs.unlink;
+        // overwrite unlink function to do the bookkeeping of project state and call original unlink.
+        _fs.unlink = function(pathname, callback){
+            _innerUnlink.call(_fs, pathname, function(err){
+                // only update cache once original was successfull
+                if (!err){
+                    delete _cache[pathname];
+                }
+                callback(err);
+            });
+        };
+
+        // original writeFile
+        var _innerWriteFile = _fs.writeFile;
+        // overwrite original writeFile and add bookkeeping of project state and call original writeFile.
+        _fs.writeFile = function(filename, data, options, callback){
+            if (typeof options === "function"){
+                callback = options;
+                options = {};
+            }
+
+            _innerWriteFile.call(_fs, filename, data, function(err){
```

Comment:
We forgot to include the possible [`options` arg](https://github.com/filerjs/filer#writeFile) here, should be:

```js
_innerWriteFile.call(_fs, filename, data, options, function(err) {
```